src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/ManageExternalAppointmentsService.kt

```diff
@@ -29,20 +32,44 @@ class ManageExternalAppointmentsService(
     prisonAppointmentRepository.findById(prisonAppointmentId).ifPresentOrElse(
       { appointment ->
         if (activitiesAppointmentsClient.isAppointmentsRolledOutAt(appointment.prisonCode)) {
-          // TODO change to call proper create method when we can establish the internal location ID.
-          activitiesAppointmentsClient.createAppointment()
-
-          log.info("EXTERNAL APPOINTMENTS: created appointment for prison appointment ID $prisonAppointmentId in activities and appointments")
+          activitiesAppointmentsClient.createAppointment(
+            prisonCode = appointment.prisonCode,
+            prisonerNumber = appointment.prisonerNumber,
+            startDate = appointment.appointmentDate,
+            startTime = appointment.startTime,
+            endTime = appointment.endTime,
+            internalLocationId = appointment.internalLocationId(),
+          )?.let { appointmentSeries ->
+            log.info("EXTERNAL APPOINTMENTS: created activities and appointments series ${appointmentSeries.id} for prison appointment $prisonAppointmentId")
+          }
         } else {
-          prisonApiClient.createAppointment()
-
-          log.info("EXTERNAL APPOINTMENTS: created appointment for prison appointment ID $prisonAppointmentId in prison api")
+          prisonApiClient.createAppointment(
+            bookingId = appointment.bookingId(),
+            appointmentType = appointment.appointmentType,
+            locationId = appointment.internalLocationId(),
+            appointmentDate = appointment.appointmentDate,
+            startTime = appointment.startTime,
+            endTime = appointment.endTime,
+            comments = appointment.comments,
+          )?.let { event ->
+            log.info("EXTERNAL APPOINTMENTS: created prison api event ${event.id} for prison appointment $prisonAppointmentId")
+          }
         }
       },
       {
         // Ignore, there is nothing we can do if we do not find the prison appointment
-        log.warn("Prison appointment with ID $prisonAppointmentId not found")
+        log.warn("EXTERNAL APPOINTMENTS: Prison appointment with ID $prisonAppointmentId not found")
       },
     )
   }
+
+  // TODO question - this should never happen but what happens if we find no location? An exception here means the event will never clean up.
+  private fun PrisonAppointment.internalLocationId() =
+    prisonApiClient.getInternalLocationByKey(prisonLocKey)?.locationId
+      ?: throw NullPointerException("Internal location id not found for prison appointment $prisonAppointmentId")
+
+  // TODO question - this should never happen but what happens if we find no booking id? An exception here means the event will never clean up.
+  private fun PrisonAppointment.bookingId() =
+    prisonerSearchClient.getPrisoner(prisonerNumber)?.bookingId?.toLong()
```

Comment:
For these, we'll need to remove messages from the queue - or have a tidying job if they happen. Something to lookout for. Steve R. did say that the location by key was 99.9% accurate - just occasionally some odd locations might not match between Locations API and NOMIS.

Reply:
Indeed.  I am not 100% sure on best course of action.  I thought by at least throwing we would be aware something is not right.  It does have the side effect of keeping things on the queue though.